frontend/src/lib/k8s/cluster.ts

```diff
@@ -430,6 +437,30 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
       return this.apiEndpoint.isNamespaced;
     }
 
+    static get readOnlyFields(): JsonPath<T>[] {
+      return this._readOnlyFields;
+    }
+
+    getEditableObject() {
+      const fieldsToRemove = this._class().readOnlyFields;
```

Comment:
We should check if fieldsToRemove is undefined as this leads to an error when trying to run the foreach later in this block.
If !fieldsToRemove, we should just return the this.jsonData.